content/threads_intro.tex

```diff
@@ -0,0 +1,73 @@
+This section specifies the interaction between \openshmem{} interfaces and the
+user threads, and also describes the routines that can be used for initializing and 
+querying the thread environment. There are four levels of threading supported by
+the \openshmem{} implementation.
+ 
+\begin{itemize}
+\item {\bf \CONST{SHMEM\_THREAD\_SINGLE}} The \openshmem{} program may not be multithreaded, 
+and all \openshmem{} interfaces are invoked by a single thread. 
+
+\item {\bf \CONST{SHMEM\_THREAD\_FUNNELED}}
+The \openshmem{} program may be multithreaded, however, the 
+program must ensure that only the main thread invokes the \openshmem{}
+interfaces, i.e, the thread that invokes either \FUNC{shmem\_init} or
+\FUNC{shmem\_init\_thread}.
+
+\item {\bf \CONST{SHMEM\_THREAD\_SERIALIZED}} 
+The \openshmem{} program may be multithreaded, however, the 
+program must ensure that only one thread invokes the \openshmem{}
+interfaces at any given instance, i.e., the \openshmem{} interfaces 
+are not invoked concurrently by multiple threads.
+
+\item {\bf \CONST{SHMEM\_THREAD\_MULTIPLE}}
+The \openshmem{} program may be multithreaded, and any 
+thread may invoke the \openshmem{} interfaces.
+\end{itemize}
+
+\hspace{-15pt}
+{\bf Semantics}
+
+\begin{itemize}
+\item
+In the \openshmem{} program, a thread calling \FUNC{shmem\_init} is designated as the main
+thread. Multiple threads may not call \FUNC{shmem\_init}. Similarly,
+\FUNC{shmem\_finalize} may only be called on the main thread.
```

Comment:
Change requested to `SHEMEM_THREAD_FUNNELED` has definition for "main thread". This paragraph may need to duplicate that text succinctly. Minutes from meeting mentioned potential for rewording this paragraph, but this paragraph needs more review.
- What is the thread level for a program calling `shmem_init`? Implementation defined?
  - Is the program `SHMEM_THREAD_SINGLE`?
  - Is the program `SHMEM_THREAD_FUNNELED`? Second sentence is not needed if that is true.
- "main thread" is only defined here. Should we also add notes to initialization and finalization routines?
  - `shmem_finalize` may need a note that it can only be called from the main thread of a multithreaded OpenSHMEM program.
  - Pinging @jdinan